File: Definitions/Game_Definitions/classes/AttackSubtypeHitPrereqCondition.lua
---@meta
---@diagnostic disable

---@class AttackSubtypeHitPrereqCondition : BaseHitPrereqCondition
---@field attackSubtype gamedataAttackSubtype
AttackSubtypeHitPrereqCondition = {}

---@return AttackSubtypeHitPrereqCondition
function AttackSubtypeHitPrereqCondition.new() return end

---@param props table
---@return AttackSubtypeHitPrereqCondition
function AttackSubtypeHitPrereqCondition.new(props) return end

---@param hitEvent gameeventsHitEvent
---@return Bool
function AttackSubtypeHitPrereqCondition:Evaluate(hitEvent) return end

---@param recordID TweakDBID|string
function AttackSubtypeHitPrereqCondition:SetData(recordID) return end

File: DecompiledGameScripts/core/gameplay/prereqs/damage/attackSubtypeHitPrereqCondition.swift

public class AttackSubtypeHitPrereqCondition extends BaseHitPrereqCondition {

  public let m_attackSubtype: gamedataAttackSubtype;

  public func SetData(recordID: TweakDBID) -> Void {
    let str: String = TweakDBInterface.GetString(recordID + t".attackSubtype", "");
    let result: gamedataAttackSubtype = IntEnum<gamedataAttackSubtype>(Cast<Int32>(EnumValueFromString("gamedataAttackSubtype", str)));
    if EnumInt(result) < 0 {
      this.m_attackSubtype = gamedataAttackSubtype.Invalid;
    } else {
      this.m_attackSubtype = result;
    };
    super.SetData(recordID);
  }

  public func Evaluate(hitEvent: ref<gameHitEvent>) -> Bool {
    let result: Bool = Equals(hitEvent.attackData.GetAttackSubtype(), this.m_attackSubtype);
    if result {
      result = this.CheckOnlyOncePerShot(hitEvent);
    };
    return this.m_invert ? !result : result;
  }
}
